src/hooks/async/contracts.ts

```diff
@@ -349,3 +353,47 @@ export function useContractFunctions(
     },
   );
 }
+
+export function useContractData(
+  contract: RequiredParam<ReturnType<typeof useContract>["contract"]>,
+  functionName: RequiredParam<string>,
+  ...args: unknown[]
+) {
+  const contractAddress = contract?.getAddress();
+  return useQueryWithNetwork(
+    cacheKeys.contract.call(contractAddress, functionName, args),
+    () => {
+      invariant(contract, "contract must be defined");
+      invariant(functionName, "function name must be provided");
+      return contract.call(functionName, ...args);
+    },
+    {
+      enabled: !!contract || !!functionName,
+    },
+  );
+}
+
+export function useContractCall(
+  contract: RequiredParam<ReturnType<typeof useContract>["contract"]>,
+) {
+  const activeChainId = useActiveChainId();
+  const contractAddress = contract?.getAddress();
+  const queryClient = useQueryClient();
+
+  return useMutation(
+    async (functionName: RequiredParam<string>, ...args: unknown[]) => {
```

Comment:
nah, keep the function name part of the hook params, only the args should be an input directly to the mutation

Reply:
also same comment about the args from above